src/interpreter/mod.rs

```diff
@@ -474,15 +474,23 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 self.frame_mut().block = target;
             }
 
-            Assert { ref cond, expected, ref msg, target, cleanup } => {
-                let actual_ptr = self.eval_operand(cond)?;
-                let actual = self.memory.read_bool(actual_ptr)?;
-                if actual == expected {
+            Assert { ref cond, expected, ref msg, target, .. } => {
+                let cond_ptr = self.eval_operand(cond)?;
+                if expected == self.memory.read_bool(cond_ptr)? {
                     self.frame_mut().block = target;
                 } else {
-                    panic!("unimplemented: jump to {:?} and print {:?}", cleanup, msg);
+                    return match *msg {
```

Comment:
No, this isn't part of the return expression. The match could be refactored though